### PR TITLE
ok now i kinda tested the ark examine

### LINKED ISSUE #32443: The time displayed for the Ark's countdown timer is 10 times larger than it should be
Same for the info tooltip.
@MoreRobustThanYou

### PR DESCRIPTION
**Pull request by ghost**
yeah yeah yeah ik gimme a PRB no update

fixes #32443

### CODE REVIEW

**Comment by Xhuis** [comment]
😂 👌 💯

**Comment by ChangelingRain** [comment]
Won't this make the examine itself inaccurate?

**Comment by MoreRobustThanYou** [comment]
@ChangelingRain wha

huh


AAAAAAAAAAAAAAAAAAAAAAA

**Comment by DoctoreMad** [comment]
----TEST YOUR PR

**Comment by nicbn** [comment]
Take the parenthesis out

**Comment by MoreRobustThanYou** [comment]
Ok, now it works.

File: code/game/objects/effects/countdown.dm [hunk 1]
```diff
@@ -108,7 +108,7 @@
 	if(!istype(G))
 		return
 	else if(G.obj_integrity && !G.purpose_fulfilled)
-		return "<div align='center' valign='middle' style='position:relative; top:0px; left:6px'>[G.get_arrival_text(FALSE)]</div>"
+		return "<div align='center' valign='middle' style='position:relative; top:0px; left:6px'>[G.get_arrival_time(FALSE)]</div>"
```
Comment (ChangelingRain) [inline code/game/objects/effects/countdown.dm]: have you checked for any other places that might possibly need fixing?

Comment (MoreRobustThanYou) [reply code/game/objects/effects/countdown.dm]: Nope, I tested this in-game, with the examine AND countdown

Comment (MoreRobustThanYou) [reply code/game/objects/effects/countdown.dm]: and also with the initial countdown, grace period, and final countdown.

Comment (ChangelingRain) [reply code/game/objects/effects/countdown.dm]: so if you haven't checked, why not?

Comment (MoreRobustThanYou) [reply code/game/objects/effects/countdown.dm]: cuz i tested all the situations in game

Comment (ChangelingRain) [reply code/game/objects/effects/countdown.dm]: that doesn't seem like valid reasoning. perhaps you should make sure you didn't _miss_ anything.

Comment (MoreRobustThanYou) [reply code/game/objects/effects/countdown.dm]: I didn't. I TESTED EVERY POSSIBLE FUCKING SITUATION FOR FUCK'S SAKE.

Comment (ChangelingRain) [reply code/game/objects/effects/countdown.dm]: I'm not asking if you tested it. I'm asking if you checked for in-code uses of it that you haven't updated.

Comment (MoreRobustThanYou) [reply code/game/objects/effects/countdown.dm]: No

File: code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm [hunk 2]
```diff
@@ -134,13 +134,22 @@
...
+			return (grace_period * 10)
+		else
+			return grace_period
 	else if(GATEWAY_RATVAR_ARRIVAL - progress_in_seconds > 0)
-		return (round(max((GATEWAY_RATVAR_ARRIVAL - progress_in_seconds) / (GATEWAY_SUMMON_RATE), 0), 1)*10)
+		if(deciseconds)
+			return (round(max((GATEWAY_RATVAR_ARRIVAL - progress_in_seconds) / (GATEWAY_SUMMON_RATE), 0), 1) * 10)
+		else
+			return round(max((GATEWAY_RATVAR_ARRIVAL - progress_in_seconds) / (GATEWAY_SUMMON_RATE), 0), 1)
 	return -10
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: not sure what this is here for.

Comment (MoreRobustThanYou) [reply code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: laziness

File: code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm [hunk 3]
```diff
@@ -134,13 +134,22 @@
...
+/obj/structure/destructible/clockwork/massive/celestial_gateway/proc/get_arrival_time(var/deciseconds = TRUE)
 	if(seconds_until_activation)
-		return (seconds_until_activation*10)
+		if(deciseconds)
+			return (seconds_until_activation * 10)
+		else
+			return seconds_until_activation
 	if(grace_period)
-		return (grace_period*10)
+		if(deciseconds)
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: this code could be compressed with use of `.`, something like this, with it being the entire block
````
/obj/structure/destructible/clockwork/massive/celestial_gateway/proc/get_arrival_time(var/deciseconds = TRUE)
 	if(seconds_until_activation)
		. = seconds_until_activation
	else if(grace_period)
		. = grace_period
	else if(GATEWAY_RATVAR_ARRIVAL - progress_in_seconds > 0)
		. = round(max((GATEWAY_RATVAR_ARRIVAL - progress_in_seconds) / (GATEWAY_SUMMON_RATE), 0), 1)
	if(deciseconds)
		. *= 10
````

**Pull request merged**